### DISCUSSION TITLE
Support for warnings

### DISCUSSION THREAD

**Discussion by nicolasnobelis** [mtkennerly/ludusavi-manifest#47] [Ideas]
GameSave Manager has a nice feature: when defining an entry, a warning can be specified that is visible in the preview list.

For instance: "this game saves take a lot of disk space" or "saves of this game may contain sensitive data"

![image](https://github.com/user-attachments/assets/76659fb5-9e00-44f3-827e-ecf11408f27f)

Does Ludusavi support something like that? I didn't find anything in the schema of the manifest.

If not, what do you think about this feature?

**Comment by nicolasnobelis** [comment]
However since the PC Gaming Wiki does not have this information, I guess it's only relevant if you curate manually the manifest :confused:

**Reply by mtkennerly** [reply]
Yeah, that would be the main challenge. PCGamingWiki does have notes for individual save paths:

* Example using italics after path: https://www.pcgamingwiki.com/wiki/Dyson_Sphere_Program#Game_data
* Example using `{{note}}` template: https://www.pcgamingwiki.com/wiki/Final_Fantasy_VII_Remake_Intergrade#Game_data

But I'm not sure how useful those kinds of notes would be generally. I think they're more focused on info like "this path applies to this game version", which helps if you're manually looking for saves, but doesn't really matter in Ludusavi since it only shows you the paths that actually exist on your system.

I'm not opposed to adding a notes field in the schema, but I'd be curious to see some specific examples of the kinds of notes that GameSave Manager has. Maybe there are other solutions to the same problem - for example, "takes a lot of space" could be addressed by some filter to flag/block backups over a certain size, so that you can automate it instead of having to remember to check the note.

**Reply by nicolasnobelis** [reply]
The two warnings in GSM I mentioned were the two I remembered from the top of my head.

I would use warnings in my manifest https://github.com/nicolasnobelis/ludusavi-psp-manifest/blob/main/manifest.yml to replace the TODOs, to make the user aware in the UI of potential issues.

See also the last paragraph of https://github.com/nicolasnobelis/ludusavi-psp-manifest/blob/main/README.md.

**Reply by mtkennerly** [reply]
That sounds reasonable :+1:

I'm thinking of just adding game-level notes for now, and we can expand it to path-specific notes later if needed. Something like:

```yaml
Some Game:
  notes:
    - message: This is a short message.
    - message: >
        This is a
        longer message.
```

Would you have any interest in an optional `level` field to control whether it gets an info or warning icon? (Info would be the default.)

**Reply by nicolasnobelis** [reply]
Great :+1: 

Regarding the level, I don't think it's quite necessary yet. Besides `info` and `warning`, does another one come to your mind?

I was ready to use mighty ASCII art like `/!\ this is my warning` :laughing:

**Reply by mtkennerly** [reply]
Updated the schema: https://github.com/mtkennerly/ludusavi-manifest/commit/516022ae90104968c25544fd5d0ab784c0a8a33e . I'll add support for it in Ludusavi for the next release.

> Besides `info` and `warning`, does another one come to your mind?

Maybe `error`, but hopefully no one is leaving confirmed errors in their manifest 😅 I'll leave it out for now.

**Reply by mtkennerly** [reply]
Here's a preview :D

> https://github.com/user-attachments/assets/12f9ee0a-0458-40e8-a6f0-5366cbd4c410

**Reply by nicolasnobelis** [reply]
Really nice !

just out of curiosity:

![image](https://github.com/user-attachments/assets/110afe94-d93e-42cc-8ea5-6c508dc5319b)

Why are the areas around the popup grayed?

**Reply by mtkennerly** [reply]
When Ludusavi came out, [Iced](https://github.com/iced-rs/iced) didn't support layering, so I did it that way instead of using a normal modal 🙃 I should probably look into updating the design now that that's possible, though.
